### PR TITLE
Rebased version of the Summer School page

### PR DESCRIPTION
**Pull request by delapuente**
This is a rebased version of #1678 on top of `main`. It needs a final review before merging. It also takes care of some style lint errors.

### CODE REVIEW

File: components/events/AccordionSection.vue [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+@Component
+export default class AccordionSection extends Vue {
+  @Prop({ type: Object }) document!: IContentDocument
+
+  @Watch('document', { immediate: true })
+  onDocumentChange (val: IContentDocument) {
+    val.body.children
+      .filter((element:any) => element.tag === 'item')
+      .forEach((element:any) => {
```
Comment (eddybrando) [inline components/events/AccordionSection.vue:23]: We should type this. Probably by destructuring `tag` and typing it.

File: components/events/AccordionSection.vue [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+import { IContentDocument } from '@nuxt/content/types/content'
+
+@Component
+export default class AccordionSection extends Vue {
+  @Prop({ type: Object }) document!: IContentDocument
+
+  @Watch('document', { immediate: true })
+  onDocumentChange (val: IContentDocument) {
+    val.body.children
+      .filter((element:any) => element.tag === 'item')
```
Comment (eddybrando) [inline components/events/AccordionSection.vue:22]: We should type this. Probably by destructuring `tag` and typing it.

File: components/events/AccordionSection.vue [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+@import "~/assets/scss/layout/_copy.scss";
+
+.accordion__section {
+  ::v-deep {
+    .bx--accordion__content {
+      @extend .copy__paragraph;
+
+      max-width: none;
+      > p {
+        margin-top: 1.43em;
```
Comment (eddybrando) [inline components/events/AccordionSection.vue:40]: Where does this number come from? Why are we not using Carbon $spacing variables?

File: components/events/summer-school/FaqSection.vue [hunk 4]
```diff
@@ -0,0 +1,323 @@
+<template>
+  <section class="faq-section">
+    <h2 class="copy__title" v-text="'Frequently asked questions'" />
```
Comment (eddybrando) [inline components/events/summer-school/FaqSection.vue:3]: Please use `v-text` when binding to a variable. When using a string literal, we usually go with:
```suggestion
    <h2 class="copy__title">
      Frequently asked questions
    </h2>
```

File: components/events/summer-school/FaqSection.vue [hunk 5]
```diff
@@ -0,0 +1,323 @@
...
+    </cv-accordion>
+  </section>
+</template>
+
+<script lang="ts">
+import Vue from 'vue'
+import { Component } from 'vue-property-decorator'
+import { SegmentData } from '~/constants/appLinks'
+
+// GeneralLink does not fit. 'label' is not necessary
```
Comment (eddybrando) [inline components/events/summer-school/FaqSection.vue:218]: We can remove this comment.

File: components/events/summer-school/FaqSection.vue [hunk 6]
```diff
@@ -0,0 +1,323 @@
...
+}
+</script>
+
+<style lang="scss" scoped>
+@import "~/assets/scss/layout/_copy.scss";
+
+.faq-section {
+  // stylelint-disable selector-pseudo-element-no-unknown
+  ::v-deep {
+  // stylelint-enable selector-pseudo-element-no-unknown
```
Comment (eddybrando) [inline components/events/summer-school/FaqSection.vue:299]: We're already disallowing this rule globally in the Stylelint config in this PR.

```suggestion
  ::v-deep {
```

File: components/events/summer-school/FaqSection.vue [hunk 7]
```diff
@@ -0,0 +1,323 @@
...
+    .bx--accordion__arrow {
+      fill: $text-color;
+    }
+
+    .bx--accordion__content {
+      @extend .copy__paragraph;
+
+      max-width: none;
+      > p + p {
+        margin-top: 1.43em;
```
Comment (eddybrando) [inline components/events/summer-school/FaqSection.vue:318]: Where does this number come from? We should try and use Carbon's $spacing variables.

File: components/ui/AppMosaic.vue [hunk 8]
```diff
@@ -21,6 +21,13 @@
             v-html="description"
           />
           <!-- estlint-enable -->
+          <AppLink
+            v-if="cta"
+            v-bind="cta"
+            class="app-mosaic__element-copy-link copy__link"
+          >
+            {{ cta.label }}
+          </AppLink>
```
Comment (eddybrando) [inline components/ui/AppMosaic.vue:30]: Nice to have: Use the `v-text` binding:

```suggestion
          <AppLink
            v-if="cta"
            v-bind="cta"
            class="app-mosaic__element-copy-link copy__link"
            v-text="cta.label"
          />
```

File: components/ui/AppMosaic.vue [hunk 9]
```diff
@@ -179,6 +188,11 @@ export default class AppMosaic extends Vue {
 
         color: $text-color-light;
       }
+
+      &-link {
+        display: flex;
+        margin-top: 1em;
```
Comment (eddybrando) [inline components/ui/AppMosaic.vue:194]: For spacing, we should use the Carbon $spacing variables.

File: constants/summerSchool2021Content.ts [hunk 10]
```diff
@@ -0,0 +1,414 @@
...
+    time: '',
+    to: 'http://qisk.it/qgss2021blog_website_source',
+    ctaLabel: 'Learn more',
+    segment: {
+      action: 'events-summer-school > header > ibm-research-blog'
+    }
+  }
+}
+
+const mosaic = {
```
Comment (eddybrando) [inline constants/summerSchool2021Content.ts:33]: I believe we should  type at least "tiles" here.

File: constants/summerSchool2021Content.ts [hunk 11]
```diff
@@ -0,0 +1,414 @@
...
+        label: 'Explore the docs',
+        segment: {
+          action: 'events-summer-school > mosaic > explore-the-docs'
+        }
+      }
+    }
+  ]
+}
+
+const week1Schedule = [
```
Comment (eddybrando) [inline constants/summerSchool2021Content.ts:91]: I believe we should add an interface to type the list elements.

File: constants/summerSchool2021Content.ts [hunk 12]
```diff
@@ -0,0 +1,414 @@
...
+  },
+  {
+    day: 'Friday, July 16',
+    topic: 'Day 5 Q&A Session',
+    speaker: '-',
+    format: 'Live Q&A'
+  }
+]
+
+const week2Schedule = [
```
Comment (eddybrando) [inline constants/summerSchool2021Content.ts:196]: I believe we should add an interface to type the list elements.

File: constants/summerSchool2021Content.ts [hunk 13]
```diff
@@ -0,0 +1,414 @@
...
+]
+
+interface AgendaSlot {
+  day: string,
+  topic: string,
+  speaker: string,
+  format: string
+}
+
+const scheduleToTableData = (slot: AgendaSlot) => ([
```
Comment (eddybrando) [inline constants/summerSchool2021Content.ts:320]: I believe we should add an interface to type the list elements.

File: constants/summerSchool2021Content.ts [hunk 14]
```diff
@@ -0,0 +1,414 @@
...
+    component: 'span',
+    styles: 'min-width: 10rem; display: inline-block; padding-top: 8px; padding-bottom: 8px',
+    data: slot.format
+  }
+])
+
+const agenda = {
+  title: 'Curriculum',
+  subtitle: '*Final agenda and syllabus subject to change',
+  weeks: [
```
Comment (eddybrando) [inline constants/summerSchool2021Content.ts:346]: I believe we should add an interface to type the list elements.

File: constants/summerSchool2021Content.ts [hunk 15]
```diff
@@ -0,0 +1,414 @@
...
+    {
+      tabName: 'Week 2',
+      tableData: week2Schedule.map(scheduleToTableData)
+    }
+  ]
+}
+
+const helpfulResources = {
+  title: 'Resources',
+  resources: [
```
Comment (eddybrando) [inline constants/summerSchool2021Content.ts:360]: I believe we should add an interface to type the list elements.

File: components/events/AccordionSection.vue (hunk 1 above)
Comment (y4izus) [reply components/events/AccordionSection.vue]: Deleted file

File: components/events/AccordionSection.vue (hunk 2 above)
Comment (y4izus) [reply components/events/AccordionSection.vue]: Deleted file

File: components/events/AccordionSection.vue (hunk 3 above)
Comment (y4izus) [reply components/events/AccordionSection.vue]: Deleted file

File: pages/events/summer-school.vue [hunk 16]
```diff
@@ -0,0 +1,231 @@
...
+<style lang="scss" scoped>
+.summer-school-page {
+  display: flex;
+  flex-direction: column;
+
+  &__section {
+    @include contained();
+
+    width: 100%;
+    margin-top: $layout-05;
```
Comment (eddybrando) [inline pages/events/summer-school.vue:157]: Try and use the $spacing variables instead of $layout.

File: pages/events/summer-school.vue [hunk 17]
```diff
@@ -0,0 +1,231 @@
...
+    margin-bottom: $layout-03;
```
Comment (eddybrando) [inline pages/events/summer-school.vue:158]: Try and use the $spacing variables instead of $layout.

File: pages/events/summer-school.vue [hunk 18]
```diff
@@ -0,0 +1,231 @@
...
+
+    @include mq($until: large) {
+      margin-bottom: $layout-01;
```
Comment (eddybrando) [inline pages/events/summer-school.vue:161]: Try and use the $spacing variables instead of $layout.

File: pages/events/summer-school.vue [hunk 19]
```diff
@@ -0,0 +1,231 @@
...
+    }
+  }
+}
+</style>
+
+<style lang="scss" scoped>
+// overrides
+.summer-school-page {
```
Comment (eddybrando) [inline pages/events/summer-school.vue:169]: This selector is already in use in the above style block. We should merge them.

File: pages/events/summer-school.vue [hunk 20]
```diff
@@ -0,0 +1,231 @@
...
+  // stylelint-disable selector-pseudo-element-no-unknown
+  ::v-deep {
+  // stylelint-enable selector-pseudo-element-no-unknown
```
Comment (eddybrando) [inline pages/events/summer-school.vue:172]: This is globally disabled in the Stylelint config in this PR.

```suggestion
  ::v-deep {
```

File: pages/events/summer-school.vue [hunk 21]
```diff
@@ -0,0 +1,231 @@
...
+    & a.bx--tabs__nav-link {
```
Comment (eddybrando) [inline pages/events/summer-school.vue:173]: Thes tab styles
```suggestion
    // TODO: Extract styles like "_checkbox" to be defined globally
    & a.bx--tabs__nav-link {
```

File: pages/events/summer-school.vue [hunk 22]
```diff
@@ -0,0 +1,231 @@
...
+      color: $text-color-light;
+      border-bottom-color: $border-color;
+
+      &:focus,
+      &:active {
+        outline: none;
```
Comment (eddybrando) [inline pages/events/summer-school.vue:180]: Not sure if this is a good idea accessibility-wise.

File: stylelint.config.js [hunk 23]
```diff
@@ -6,6 +6,14 @@ module.exports = {
   // https://stylelint.io/user-guide/configuration
   rules: {
     'at-rule-no-unknown': null,
-    'scss/at-rule-no-unknown': true
+    'scss/at-rule-no-unknown': true,
+    // To ignore valid ::v-deep selector:
+    // https://github.com/bencodezen/vue-enterprise-boilerplate/issues/190
```
Comment (eddybrando) [inline stylelint.config.js:11]: I believe we can remove this comments. The code should already be explicit enough.

```suggestion
```

File: pages/events/summer-school.vue (hunk 16 above)
Comment (y4izus) [reply pages/events/summer-school.vue:157]: I think it's correct to use $layout here, is for the separation between sections

File: pages/events/summer-school.vue (hunk 17 above)
Comment (y4izus) [reply pages/events/summer-school.vue:158]: Same 👆

File: pages/events/summer-school.vue (hunk 18 above)
Comment (y4izus) [reply pages/events/summer-school.vue:161]: Same 👆

**Pull request merged**